File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] product = new int[nums.length];
        int[] right = new int[nums.length];

        product[0] = 1;
        right[nums.length - 1] = 1;

        for(int i = nums.length - 2; i >=0 ; i--){
            right[i] = nums[i+1] * right[i+1];
        }

        for(int i = 1; i < nums.length; i++){
            product[i] = nums[i-1] * product[i-1];
        }

        for(int i = 0; i < nums.length; i++){
            product[i] *= right[i];
        }

        return product; 
    }
}  

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:

        if len(nums) == 0:
            return False

        reach = [False] * len(nums)

        reach[0] = True

        for index, num in enumerate(nums):
            if not reach[index]:
                continue
            
            for i in range(num):
                if (index + i + 1) < len(nums):
                    reach[index + i + 1] = True
                else:
                    break
        
        return reach[-1]

        